extensions/objects/templates/position-of-terms-editor.directive.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for position of terms editor.
+ */
+
+// Every editor directive should implement an alwaysEditable option. There
+// may be additional customization options for the editor that should be passed
+// in via initArgs.
+
+angular.module('oppia').directive('positionOfTermsEditor', [
+  function() {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {
+        value: '='
+      },
+      template: require('./position-of-terms-editor.directive.html'),
+      controllerAs: '$ctrl',
+      controller: ['$scope', function($scope) {
+        var ctrl = this;
+        ctrl.$onInit = function() {
+          $scope.$watch('$ctrl.localValue.position', function() {
+            ctrl.value = ctrl.localValue.position.name;
+          });
+          ctrl.alwaysEditable = true;
+
+          ctrl.positionOfTerms = [{
+            name: 'lhs',
```

Comment:
If these are going to be used elsewhere, maybe in directives that check this, have it as constants in constants.ts?

Reply:
This entire list of objects won't be used elsewhere. This directive will provide the 'name' of the position, i.e., 'lhs', 'rhs', etc., so while validating I'd just be checking the value provided against these 4 names. There's no need to make the entire thing a global constant.
Also, note that this is the convention used in other places as well. [Ref1](https://github.com/oppia/oppia/blob/develop/extensions/objects/templates/graph-property-editor.directive.ts), [Ref2](https://github.com/oppia/oppia/blob/develop/extensions/objects/templates/logic-error-category-editor.directive.ts).